### DISCUSSION TITLE
Insufficient webhooks when client disconnects from EMQ node

### DISCUSSION THREAD

**Discussion by hiimdoublej-swag** [emqx/emqx#6392] [Q&A] [answered]
When a client disconnects from a cluster,
it seems that it's only sending 1 `client.disconnected` webhook.
But I was expecting  `N * session.unsubscribed` + `1 * client.disconnected` webhooks (N being the number of topics that the client is subscribed to at the time), since the client would be considered unsubscribed from all the topics as well ?
Isn't this the desired behavior ?

EDIT: @qzhuyan You converted this into a discussion, may I ask you to help answering this ?
Thanks

**Comment by hiimdoublej-swag** [comment]
Gentle ping @qzhuyan

**Comment by hiimdoublej-swag** [comment]
Gentle ping @zmstone , is it possible that we have this feature ? `client.unsubscribed` events whenever a `client.disconnected` happens.

**Comment by zmstone** [comment]
A disconnection doesn’t trigger unsubscribes if it’s a persistent session.

for non persistent sessions, EMQ X can be made configurable to trigger implicit unsubscribe events triggered from the broker side.

agree this is a legit requirement.

**Comment by zmstone** [comment]
@hiimdoublej-swag could you create a “feature request “ issue for us to follow up?

**Reply by hiimdoublej-swag** [reply]
For the record this was a issue at first, but I guess I can make another one in the form of a feature request. I'll post a link here.

**Reply by zmstone** [reply]
Sorry for the back-n-forth

**Comment by hiimdoublej-swag** [comment, marked as answer]
@zmstone I've submitted a related issue (#6548), can you help triage that ? I'll mark this discussion as answered.
